scg-system/src/main/java/io/telicent/backup/services/DatasetBackupService.java

```diff
@@ -0,0 +1,276 @@
+            }
+        }
+        response.set("datasets", datasetNodes);
+        return response;
+    }
+
+    ObjectNode backupTDB(DatasetGraph dsg, String backupPath, String datasetName) {
+        ObjectNode node = MAPPER.createObjectNode();
+        String tdbBackupPath = backupPath + "/TDB/";
+        if (!createPathIfNotExists(tdbBackupPath)) {
+            node.put("reason", "Cannot create backup directory: " + tdbBackupPath);
+            node.put("success", false);
+            return node;
+        }
+        try {
+            executeBackupTDB(dsg, tdbBackupPath + datasetName+ "_backup"); // .nq.gz will be added automatically.
+            node.put("folder", tdbBackupPath);
+            node.set("files", populateNodeFromDir(tdbBackupPath));
+            node.put("success", true);
+        } catch (RuntimeException e) {
+            node.put("reason", e.getMessage());
+            node.put("success", false);
+        }
+        return node;
+    }
+
+    /**
+     * Wrapping actual back-up call to aid with testings sake.
+     * @param dsg Dataset to backup
+     * @param backupFile the file to back up to
+     */
+    @ExcludeFromJacocoGeneratedReport
+    void executeBackupTDB(DatasetGraph dsg, String backupFile) {
+        Backup.backup(dsg, dsg, backupFile);
+    }
+
+    ObjectNode backupLabelStore(DatasetGraph dsg, String backupPath) {
+        ObjectNode node = MAPPER.createObjectNode();
+        if (dsg instanceof DatasetGraphABAC abac) {
+            LabelsStore labelsStore = abac.labelsStore();
+            if (labelsStore instanceof LabelsStoreRocksDB rocksDB) {
+                String labelBackupPath = backupPath + "/labels/";
+                if (!createPathIfNotExists(labelBackupPath)) {
+                    node.put("reason", "Cannot create label backup directory: " + labelBackupPath);
+                    node.put("success", false);
+                } else {
+                    try {
+                        executeBackupLabelStore(rocksDB, labelBackupPath);
+                        node.put("folder", labelBackupPath);
+                        node.set("files", populateNodeFromDir(labelBackupPath));
+                        node.put("success", true);
+                    } catch (RuntimeException e) {
+                        node.put("reason", e.getMessage());
+                        node.put("success", false);
+                    }
+                }
+            } else {
+                node.put("reason", "No Label Store to back up (not RocksDB)");
+                node.put("success", false);
+            }
+        } else {
+            node.put("reason", "No Label Store to back up (not ABAC)");
+            node.put("success", false);
+        }
+        return node;
+    }
+
+    @ExcludeFromJacocoGeneratedReport
+    void executeBackupLabelStore(LabelsStoreRocksDB rocksDB, String labelBackupPath) {
+        // Disabling for the time-being
+//        rocksDB.backup(labelBackupPath);
```

Comment:
Looks like you fixed this in your PR telicent-oss/rdf-abac#74 - can this be restored once you have a new RDF-ABAC release made?

Reply:
The commit will come once the release goes through
https://github.com/telicent-oss/rdf-abac/pull/75